File: Lyft Backend Software Engineering/car.py
from serviceable import Serviceable

# Car class represents a car and implements the Serviceable interface.
class Car(Serviceable):
    # Constructor initializes the car with an engine, battery, and tires.
    def __init__(self, engine, battery, tires):
        self.engine = engine
        self.battery = battery
        self.tires = tires

    # needs_service method checks if the car needs service based on its engine, battery, and tires.
    def needs_service(self):
        return self.engine.needs_service() or self.battery.needs_service() or self.tires.needs_service()

File: Softstraw Back-End Engineer/car.java
import serviceable.Serviceable;

// Car class represents a car and implements the Serviceable interface.
public class Car implements Serviceable {

    // Fields to store the car's engine, battery, and tires.
    private Engine engine;
    private Battery battery;
    private Tires tires;

    // Constructor initializes the car with an engine, battery, and tires.
    public Car(Engine engine, Battery battery, Tires tires) {
        this.engine = engine;
        this.battery = battery;
        this.tires = tires;
    }

    // needsService method checks if the car needs service based on its engine, battery, and tires.
    @Override
    public boolean needsService() {
        return engine.needsService() || battery.needsService() || tires.needsService();
    }
}
